src/float.cr

```diff
@@ -148,13 +149,33 @@ struct Float32
     Printer.print(self, io)
   end
 
-  def hash
-    unsafe_as(Int32)
-  end
-
   def clone
     self
   end
+
+  include Number::HashNormalize
+
+  def hash_normalize
+    float_normalize_wrap do
+      {% if flag?(:x86) || flag?(:x86_64) || flag(:arm) || flag(:aarch64) %}
+	# it should work on every architecture where endianess of Float32 and Int32
```

Comment:
I see: i used tabs unintentionally instead of spaces.
Fixed that, and reduced relying on macroses.